workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/InputSpecValidator.java

```diff
@@ -0,0 +1,126 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import java.io.File;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.constants.ToolConstants;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.exception.ControllerErrorCodes;
+import io.hyscale.controller.model.HyscaleCommandSpec;
+import io.hyscale.controller.util.ServiceProfileUtil;
+
+@Component
+public class InputSpecValidator implements Validator<HyscaleCommandSpec> {
+    
+    private static final Logger logger = LoggerFactory.getLogger(InputSpecValidator.class);
+
+    @Autowired
+    private ServiceSpecFileValidator serviceSpecFileValidator;
+
+    @Autowired
+    private ServiceSpecSchemaValidator serviceSpecSchemaValidator;
+
+    @Autowired
+    private ProfileFileValidator profileFileValidator;
+
+    @Autowired
+    private ProfileSpecSchemaValidator profileSpecSchemaValidator;
+
+    /**
+     * Steps:
+     * FileValidator for service spec
+     * SchemaValidator - Service Spec
+     * 
+     * 1. If profile Name given:
+     *      Process - getProfileFiles - return List<ProfileFile> throws exception
+     *          ServiceSpec File to Model
+     *          Get All profile Files - 
+     *      
+     * FileValidator for profiles    
+     * SchemaValidator Profiles
+     */
+    @Override
+    public boolean validate(HyscaleCommandSpec commandSpec) throws HyscaleException {
+        if (commandSpec == null) {
+            return false;
+        }
+        List<File> serviceSpecs = commandSpec.getServiceSpecFiles();
+
+        if (isInvalid(serviceSpecFileValidator, serviceSpecs)) {
+            return false;
+        }
+
+        if (isInvalid(serviceSpecSchemaValidator, serviceSpecs)) {
+            return false;
+        }
+
+        List<File> profiles = commandSpec.getProfileFiles();
+        if (profiles == null && commandSpec.getProfileName() == null) {
+            return true;
+        }
+
+        if (profiles == null && commandSpec.getProfileName() != null) {
+            profiles = ServiceProfileUtil.getProfilesFromName(serviceSpecs, commandSpec.getProfileName());
+            commandSpec.setProfileFiles(profiles);
```

Comment:
Can be processed before this validation and then hyscaleCommandSpec can have that ?
hyscaleCommandSpec will just have profiles in such case.

Reply:
What if we use the same in the converter of the profile name option. You might get many profiles but we would process only for those that match with servicespecs ?
 
It's like this:
1. -P converter: read all profiles found from the search directory
2. Command's call method : filter with the servicespecs.
3. Build Hyscale command spec with the filtered profiles and use them for validation.

Reply:
Few issues in that approach:
1. In Profile converter we don't have the directory information. In case we call the converter inside call method. We need the service name information which we are only fetching after validating service spec.

Reply:
-P Profile option does not indicate of services, probably the filtering of services will come later into picture that is during Effective service spec.

Reply:
Will discuss on this on the call.